### PR TITLE
Track Read and Write API usage

### PR DESCRIPTION
**Pull request by prashastia**
Populate User Agent Header and Trace ID.

/gcbrun

### CODE REVIEW

File: flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/source/split/SplitDiscoverer.java [hunk 1]
```diff
@@ -89,6 +89,8 @@ public static List<String> discoverSplits(
                     ReadSession.newBuilder()
                             .setTable(srcTable)
                             .setDataFormat(format)
+                            //                            .setTraceId(TRACE_ID)  // Set Trace ID for
+                            // Read API once support is added.
```
Comment (jayehwhyehentee) [inline flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/source/split/SplitDiscoverer.java:93]: Please remove. Create a bug for tracking this internally.

File: flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesImpl.java [hunk 2]
```diff
@@ -185,17 +189,14 @@ public void close() {
...
 
         private StorageWriteClientImpl(CredentialsOptions options) throws IOException {
             BigQueryWriteSettings.Builder settingsBuilder =
                     BigQueryWriteSettings.newBuilder()
                             .setCredentialsProvider(
                                     FixedCredentialsProvider.create(options.getCredentials()))
+                            .setHeaderProvider(USER_AGENT_HEADER_PROVIDER)
                             .setTransportChannelProvider(
                                     BigQueryReadSettings.defaultGrpcTransportProviderBuilder()
                                             .setHeaderProvider(USER_AGENT_HEADER_PROVIDER)
```
Comment (jayehwhyehentee) [inline flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesImpl.java:202]: Why are we setting the user-agent twice in the same client?

File: flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesImpl.java [hunk 3]
```diff
@@ -122,17 +129,14 @@ public void cancel() {
...
 
         private StorageReadClientImpl(CredentialsOptions options) throws IOException {
             BigQueryReadSettings.Builder settingsBuilder =
                     BigQueryReadSettings.newBuilder()
                             .setCredentialsProvider(
                                     FixedCredentialsProvider.create(options.getCredentials()))
+                            .setHeaderProvider(USER_AGENT_HEADER_PROVIDER)
                             .setTransportChannelProvider(
                                     BigQueryReadSettings.defaultGrpcTransportProviderBuilder()
                                             .setHeaderProvider(USER_AGENT_HEADER_PROVIDER)
```
Comment (jayehwhyehentee) [inline flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesImpl.java:142]: Why are we setting the user-agent twice in the same client?

File: flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesImpl.java (hunk 2 above)
Comment (prashastia) [reply flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesImpl.java:202]: The existing one is set via the transport channel. This User Agent is not parsed by either of the API's hence we need to se the User Agent via the `setHeaderProvider()`

Comment (jayehwhyehentee) [reply flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesImpl.java:202]: I understand that you are setting the header in a different way now, so my question is do we still need the earlier process of setting it in the transport channel?

**Review by jayehwhyehentee — APPROVED** [review]

File: flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesImpl.java [hunk 4]
```diff
@@ -122,20 +129,16 @@ public void cancel() {
...
-                        "user-agent", "Apache_Flink_Java/" + FlinkVersion.current().toString());
-
         private final BigQueryReadClient client;
 
         private StorageReadClientImpl(CredentialsOptions options) throws IOException {
             BigQueryReadSettings.Builder settingsBuilder =
                     BigQueryReadSettings.newBuilder()
                             .setCredentialsProvider(
                                     FixedCredentialsProvider.create(options.getCredentials()))
+                            .setHeaderProvider(USER_AGENT_HEADER_PROVIDER)
```
Comment (Shailesh-Tripathi) [inline flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesImpl.java:139]: Just to clarify, this header will be provided in both CreateReadSession and ReadRows call, right?

**Review by Shailesh-Tripathi — APPROVED** [review]
LGTM

File: flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesImpl.java (hunk 4 above)
Comment (prashastia) [reply flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/services/BigQueryServicesImpl.java:139]: Yes. The same BigQueryReadClient obtained from these settings is utilized for CreateReadSession and ReadRows call.

**Pull request merged**